ra/deployment/src/main/java/io/quarkus/artemis/jms/ra/deployment/DevServicesArtemisProcessor.java

```diff
@@ -0,0 +1,362 @@
+                    name);
+            return null;
+        }
+
+        // Check if quarkus.artemis.url is set
+        String urlPropertyName = getUrlPropertyName(name);
+        if (ConfigUtils.isPropertyPresent(urlPropertyName)) {
+            LOGGER.debugf(
+                    "Not starting dev services for ActiveMQ Artemis and configuration %s, the quarkus.artemis.url is configured.",
+                    name);
+            return null;
+        }
+
+        if (!dockerStatusBuildItem.isContainerRuntimeAvailable()) {
+            LOGGER.warn(
+                    "Docker isn't working, please configure the ActiveMQ Artemis Url property (quarkus.artemis.url).");
+            return null;
+        }
+
+        final Optional<ContainerAddress> maybeContainerAddress = artemisContainerLocator.locateContainer(config.serviceName,
+                config.shared,
+                launchMode.getLaunchMode());
+
+        // Starting the broker
+        String containerName = "ActiveMQ-Artemis " + name;
+        Supplier<RunningDevService> defaultArtemisBrokerSupplier = () -> {
+            ArtemisContainer container = new ArtemisContainer(
+                    DockerImageName.parse(config.imageName),
+                    config.fixedExposedPort,
+                    config.user,
+                    config.password,
+                    config.extraArgs);
+            container.withReuse(config.reuse);
+
+            ConfigureUtil.configureSharedNetwork(container, "artemis");
+
+            if (config.serviceName != null) {
+                container.withLabel(DevServicesArtemisProcessor.DEV_SERVICE_LABEL, config.serviceName);
+            }
+
+            timeout.ifPresent(container::withStartupTimeout);
+
+            container.start();
+            var configuration = createConnectionParameters(urlPropertyName, container.getHost(), container.getPort());
+            return new RunningDevService(
+                    containerName,
+                    container.getContainerId(),
+                    container::close,
+                    configuration);
+
+        };
+
+        return maybeContainerAddress
+                .map(containerAddress -> new RunningDevService(
+                        containerName,
+                        containerAddress.getId(),
+                        null,
+                        createConnectionParameters(urlPropertyName, containerAddress.getHost(), containerAddress.getPort())))
+                .orElseGet(defaultArtemisBrokerSupplier);
+    }
+
+    private static Map<String, String> createConnectionParameters(String urlPropertyName, String host, int port) {
+        return Map.of(urlPropertyName,
+                String.format("host=%s;port=%d;protocols=CORE", host,
+                        port),
+                "protocol-manager-factory",
+                "org.apache.activemq.artemis.core.protocol.hornetq.client.HornetQClientProtocolManagerFactory");

```

Comment:
Just curious that `protocol-manager-factory` could be configurable? Or only with HornetQ protocol, as far as I know it could be deprecated.